source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp

```diff
@@ -0,0 +1,1225 @@
+        GLenum colorList[1] = { GL_COLOR_ATTACHMENT0 };
+        glDrawBuffers(1, colorList);
+    }
+    // Unbind frame buffer and route nowhere.
+    else
+    {
+        glBindFramebuffer(GL_FRAMEBUFFER, UNDEFINED_OPENGL_RESOURCE_ID);
+        glDrawBuffer(GL_NONE);
+    }
+    return true;
+}
+
+void GlslValidator::deleteProgram()
+{
+    if (_programId > 0)
+    {
+        GLUtilityContext* context = GLUtilityContext::get();
+        if (context && context->makeCurrent())
+        {
+            glDeleteObjectARB(_programId);
+        }
+        _programId = UNDEFINED_OPENGL_RESOURCE_ID;
+    }
+
+    // Program deleted, so also clear cached input lists
+    clearInputLists();
+}
+
+unsigned int GlslValidator::createProgram()
+{
+    ShaderValidationErrorList errors;
+    const std::string errorType("GLSL program creation error.");
+
+    GLUtilityContext* context = GLUtilityContext::get();
+    if (!context)
+    {
+        errors.push_back("No valid OpenGL context to create program with.");
+        throw ExceptionShaderValidationError(errorType, errors);
+
+    }
+    if (!context->makeCurrent())
+    {
+        errors.push_back("Cannot make OpenGL context current to create program.");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    deleteProgram();
+
+    if (!haveValidStages())
+    {
+        errors.push_back("An invalid set of stages has been provided.");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    GLint GLStatus = GL_FALSE;
+    int GLInfoLogLength = 0;
+
+    unsigned int stagesBuilt = 0;
+    unsigned int desiredStages = 0;
+    for (unsigned int i = 0; i < HwShader::NUM_STAGES; i++)
+    {
+        if (_stages[i].length())
+            desiredStages++;
+    }
+
+    // Create vertex shader
+    GLuint vertexShaderId = UNDEFINED_OPENGL_RESOURCE_ID;
+    std::string &vertexShaderSource = _stages[HwShader::VERTEX_STAGE];
+    if (vertexShaderSource.length())
+    {
+        vertexShaderId = glCreateShader(GL_VERTEX_SHADER);
+
+        // Compile vertex shader
+        const char* vertexChar = vertexShaderSource.c_str();
+        glShaderSource(vertexShaderId, 1, &vertexChar, NULL);
+        glCompileShader(vertexShaderId);
+
+        // Check Vertex Shader
+        glGetShaderiv(vertexShaderId, GL_COMPILE_STATUS, &GLStatus);
+        if (GLStatus == GL_FALSE)
+        {
+            errors.push_back("Error in compiling vertex shader:");
+            glGetShaderiv(vertexShaderId, GL_INFO_LOG_LENGTH, &GLInfoLogLength);
+            if (GLInfoLogLength > 0)
+            {
+                std::vector<char> vsErrorMessage(GLInfoLogLength + 1);
+                glGetShaderInfoLog(vertexShaderId, GLInfoLogLength, NULL,
+                    &vsErrorMessage[0]);
+                errors.push_back(&vsErrorMessage[0]);
+            }
+        }
+        else
+        {
+            stagesBuilt++;
+        }
+    }
+
+    // Create fragment shader
```

Comment:
When we have more shader stages this can be rejigged a bit to allow for stage replacement if performance becomes an issues.